### PR TITLE
Add a unique report_group_id to found groups in reports

### PR DESCRIPTION
**Pull request by bmedx**
**Description:** Adds ability to link found group members in reports

### CODE REVIEW

File: code_annotations/base.py [hunk 1]
```diff
@@ -352,6 +352,20 @@ def _check_results_choices(self, annotation):
                 'No choices found for "{}". Expected one of {}.'.format(token, self.config.choices[token])
             )
 
+    def _get_group_children(self):
```
Comment (bmedx) [inline code_annotations/base.py]: This is just a refactor since we use it in two places now.

File: code_annotations/base.py [hunk 2]
```diff
@@ -446,6 +449,12 @@ def check_results(self, all_results):
                             current_group, token, annotation['line_number'])
                         )
 
+                # If we have all members, this group is done
```
Comment (bmedx) [inline code_annotations/base.py]: Moved this out here to catch the cases where a group only has one member. Not sure why you'd do that, but it should work.

File: code_annotations/cli.py [hunk 3]
```diff
@@ -126,14 +126,6 @@ def django_find_annotations(
 def static_find_annotations(config_file, source_path, report_path, verbosity, lint, report):
     """
     Subcommand to find annotations via static file analysis.
-
```
Comment (bmedx) [inline code_annotations/cli.py]: Ned rightfully mentioned that this is unnecessary in Click commands.

**Comment by nedbat** [comment]
I;m having difficulty getting the technical mindset to review this code except to say: trailing commas are a good thing, and we should use them more :)

File: tests/test_base.py [hunk 4]
```diff
@@ -75,3 +73,100 @@ def test_bad_coverage_targets(test_config, expected_message):
...
+                'line_number': 2,
+                'annotation_token': 'foo',
+                'annotation_data': 'Group id 1',
+                'expected_group_id': 4
+            }
+        ]
+
+    results = search._format_results_for_report(fake_results)  # pylint: disable=protected-access
+
+    for filename in fake_results:
```
Comment (macdiesel) [inline tests/test_base.py]: Nit: This is a bit to parse, want to want to break it up or add some comments.

**Pull request merged**